packages/app/src/cli/models/app/loader.ts

```diff
@@ -319,15 +343,18 @@ class AppLoader<TConfig extends AppConfiguration, TModuleSpec extends ExtensionS
     const extensions = await this.loadExtensions(directory, configuration)
 
     const packageJSONPath = joinPath(directory, 'package.json')
-    const name = await loadAppName(directory)
-    const nodeDependencies = await getDependencies(packageJSONPath)
-    const packageManager = await getPackageManager(directory)
+
+    // These don't need to be processed again if the app is being reloaded
+    const name = this.previousApp?.name ?? (await loadAppName(directory))
+    const nodeDependencies = this.previousApp?.nodeDependencies ?? (await getDependencies(packageJSONPath))
```

Comment:
Checking on this one: If I generate extension whilst dev is running, are we expecting that new extension to be picked up? It may involve dependency changes

Reply:
If using workspaces the dependency will be in the extension's package.json, but I guess it is true for non-worksapce apps.
In any case, just checked and `nodeDependencies` is not used for anything